NitroxServer/Listener.cs

```diff
@@ -52,29 +52,51 @@ public void ClientAccepted(IAsyncResult ar)
             Console.WriteLine("New client connected");
 
             Socket socket = (Socket)ar.AsyncState;
-            
+
             Connection connection = new Connection(socket.EndAccept(ar)); // TODO: Will this throw an error if timed correctly?
+            connection.BeginReceive(new AsyncCallback(dropPacketsUntilAuthentication));
             connection.BeginReceive(new AsyncCallback(DataReceived));
 
             socket.BeginAccept(new AsyncCallback(ClientAccepted), socket);
         }
 
-        public void DataReceived(IAsyncResult ar)
+        private void dropPacketsUntilAuthentication(IAsyncResult ar)
         {
             Connection connection = (Connection)ar.AsyncState;
-            
-            foreach(Packet packet in connection.GetPacketsFromRecievedData(ar))
-            {
-                Player player = GetPlayer(packet, connection);
-                connection.PlayerId = player.Id;
-                UpdatePlayerPosition(player, packet);
+            bool authenticated = false;
```

Comment:
Is there way to consolidate this into a single function?  Maybe refactoring authenticated into a member of connection.